### PR TITLE
HBASE-22574 hbase-filesystem does not build against HBase 1

### PR DESCRIPTION
**Pull request by apurtell**

### CODE REVIEW

File: hbase-oss/src/test/java/org/apache/hadoop/hbase/oss/sync/TestTreeLockManager.java [hunk 1]
```diff
@@ -16,14 +16,14 @@
  * limitations under the License.
  */
 
-package org.apache.hadoop.hbase.oss;
+package org.apache.hadoop.hbase.oss.sync;
```
Comment (apurtell) [inline hbase-oss/src/test/java/org/apache/hadoop/hbase/oss/sync/TestTreeLockManager.java]: Unrelated change, I can drop it if you like, but I found this test to be in the wrong package.

**Comment by apurtell** [comment]
I opened a PR before realizing this won't be so easy. Please wait until I push a revision.

**Comment by apurtell** [comment]
With `mvn -Dhbase.profile=1 dependency:tree | grep hbase`

    ...
    [INFO] org.apache.hbase.filesystem:hbase-filesystem:pom:1.0.0-alpha1
    [INFO] +- org.apache.hbase.thirdparty:hbase-shaded-miscellaneous:jar:2.2.1:compile
    [INFO] +- org.apache.hbase:hbase-testing-util:jar:1.4.10:test
    ...

With `mvn -Dhbase.profile=1 clean install`

    ...
    [WARNING] Tests run: 160, Failures: 0, Errors: 0, Skipped: 21
    ...
    [INFO] BUILD SUCCESS

With `mvn dependency:tree | grep hbase`

    ...
    [INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ hbase-filesystem ---
    [INFO] org.apache.hbase.filesystem:hbase-filesystem:pom:1.0.0-alpha1
    [INFO] +- org.apache.hbase:hbase-zookeeper:jar:2.1.4:test
    ...

With `mvn clean install`

    ...
    [WARNING] Tests run: 160, Failures: 0, Errors: 0, Skipped: 21
    ...
    [INFO] BUILD SUCCESS

File: pom.xml [hunk 2]
```diff
@@ -126,5 +128,79 @@
...
+	  <version>${hbase1.version}</version>
+	  <type>test-jar</type>
+	  <scope>test</scope>
+	</dependency>
+	<dependency>
+	  <groupId>org.apache.hadoop</groupId>
+	  <artifactId>hadoop-minikdc</artifactId>
+	  <version>${hadoop.version}</version>
+	  <exclusions>
+	    <exclusion>
```
Comment (apurtell) [inline pom.xml]: Exclusion needed here to avoid a dependency resolution problem with -Dhadoop.version=2.9.2

**Comment by liuml07** [comment]
I tested and following commandscould pass all tests (skipped 21).

```
mvn clean install -Dhbase.profile=1
mvn clean install -Dhbase.profile=1 -Dhadoop.version=2.9.2
```

File: hbase-oss/src/test/java/org/apache/hadoop/hbase/oss/sync/TestTreeLockManager.java (hunk 1 above)
Comment (wchevreuil) [reply hbase-oss/src/test/java/org/apache/hadoop/hbase/oss/sync/TestTreeLockManager.java]: Nice catch! I think it's fine to include it within this work.

File: hbase-oss/pom.xml [hunk 3]
```diff
@@ -311,6 +318,7 @@
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-mapreduce-client-jobclient</artifactId>
       <version>${hadoop.version}</version>
+      <type>test-jar</type>
       <scope>test</scope>
     </dependency>
     <dependency>
```
Comment (wchevreuil) [inline hbase-oss/pom.xml]: Adding these allowed me to get build issues [reported here](https://jira.apache.org/jira/browse/HBASE-22574?focusedCommentId=16862654&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16862654) sorted.

File: pom.xml [hunk 4]
```diff
@@ -93,15 +95,6 @@
...
-      <dependencies>
-        <!-- TestHBOSSContractDistCp hard-codes some HDFS classes in Hadoop 2 -->
-        <dependency>
-          <groupId>org.apache.hadoop</groupId>
-          <artifactId>hadoop-hdfs</artifactId>
-          <version>${hadoop.version}</version>
-          <scope>test</scope>
-        </dependency>
-      </dependencies>
     </profile>
```
Comment (wchevreuil) [inline pom.xml]: Moved to general test dependencies on hboss pom module, as it was then required for both hadoop profiles (see comment above).

**Comment by wchevreuil** [comment]
Had pushed some minor changes in both pom files. These were needed to get rid of the build problems [mentioned here](https://jira.apache.org/jira/browse/HBASE-22574?focusedCommentId=16862654&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16862654).

**Comment by apurtell** [comment]
Thanks @wchevreuil .

**Comment by apurtell** [comment]
Are we good to go here?

File: pom.xml [hunk 5]
```diff
@@ -126,5 +119,79 @@
...
+      </activation>
+      <properties>
+        <hbase.version>${hbase2.version}</hbase.version>
+      </properties>
+      <!-- For testing against ZK -->
+      <dependencies>
+	<dependency>
+	  <groupId>org.apache.hbase</groupId>
+	  <artifactId>hbase-zookeeper</artifactId>
+	  <version>${hbase2.version}</version>
```
Comment (mackrorysd) [inline pom.xml]: There are tabs being used here and spaces everywhere else. I'm fine with either, but let's be consistent.

File: pom.xml [hunk 6]
```diff
@@ -126,5 +119,79 @@
         <hadoop.version>${hadoop3.version}</hadoop.version>
       </properties>
     </profile>
+    <profile>
+      <!-- default HBase profile -->
+      <id>hbase2</id>
+      <activation>
+        <property>
+          <name>!hbase.profile</name>
```
Comment (mackrorysd) [inline pom.xml]: It would be nice if -Dhbase.profile=2 worked and was the default. Similarly, -Dhadoop.profile=3 is the default but specifying it works.

**Comment by apurtell** [comment]
I rebased the patch here on https://jira.apache.org/jira/secure/attachment/12971822/0001-HBASE-22574-hbase-filesystem-does-not-build-against-.patch

**Comment by apurtell** [comment]
Tested with

    mvn clean verify -Dhbase.profile=1 -Dhadoop.profile=2
    mvn clean verify -Dhbase.profile=1 -Dhadoop.profile=3
    mvn clean verify -Dhbase.profile=2 -Dhadoop.profile=2
    mvn clean verify -Dhbase.profile=2 -Dhadoop.profile=3
    mvn clean verify

Looks good here

**Pull request merged**